agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -35,63 +36,44 @@
 import javax.net.ssl.SSLSocketFactory;
 import org.checkerframework.checker.nullness.qual.Nullable;
 import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class NetworkFriendlyExceptions {
-  public static final List<String> EXPECTED_CIPHERS =
-      Arrays.asList(
-          "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
-          "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
-          "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
-          "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256");
+
+  private static final List<FriendlyExceptionDetector> DETECTORS;
+  private static final Logger logger = LoggerFactory.getLogger(NetworkFriendlyExceptions.class);
+
+  static {
+    DETECTORS = new ArrayList<>();
+    try {
+      // Note this order is important to determine the right exception!
+      // For example SSLHandshakeException extends IOException
+      DETECTORS.add(SSLExceptionDetector.create());
+      DETECTORS.add(UnknownHostExceptionDetector.create());
+      DETECTORS.add(CipherExceptionDetector.create());
+    } catch (NoSuchAlgorithmException e) {
+      logger.debug(e.getMessage(), e);
+    }
```

Comment:
let's wrap only this one line in the catch block, otherwise someone may add detector after it which would not get added if exception thrown